app/plugin_api.go

```diff
@@ -914,3 +914,59 @@ func (api *PluginAPI) PluginHTTP(request *http.Request) *http.Response {
 	api.app.ServeInterPluginRequest(responseTransfer, request, api.id, destinationPluginId)
 	return responseTransfer.GenerateResponse()
 }
+
+func (api *PluginAPI) CreateSlashCommand(cmd *model.Command) (*model.Command, *model.AppError) {
+	cmd.CreatorId = ""
+	cmd.PluginId = api.id
+
+	cmd, appErr := api.app.CreateCommand(cmd)
+
+	return cmd, appErr
+}
+
+func (api *PluginAPI) ListSlashCommands(teamID string, customOnly bool) ([]*model.Command, *model.AppError) {
+	var commands []*model.Command
+	var appErr *model.AppError
+
+	if customOnly {
+		commands, appErr = api.app.ListTeamCommands(teamID)
+	} else {
+		commands, appErr = api.app.ListAllCommands(teamID, api.app.T)
```

Comment:
Off topic for this review, but the presence of `api.app.T` at this layer surprises me. Do we support translating the built-in slash commands to the language of the requesting user? How should that translate into a plugin request? Do we "force english"?

Reply:
Looks like you're right: it's for built-in commands for translating the `AutoCompleteDesc` and `DisplayName`, eg: https://github.com/mattermost/mattermost-server/blob/a63684fcb5e3ba7b7522b35c29a4cb27779ba823/app/command_dnd.go#L26-L33
I don't know what language the plugin would take... the system default?